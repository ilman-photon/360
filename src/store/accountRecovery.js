import { createSlice, createAsyncThunk } from "@reduxjs/toolkit";
import { Api } from "../pages/api/api";

export const fetchPatientAccount = createAsyncThunk(
  "accountRecovery/fetchPatientAccount",
  async ({ keyword }) => {
    const api = new Api();
    const url = `/ecp/accountRecovery/getPatientDetails/${keyword}`;
    return api.getResponse(url, null, "get");
  }
);

export const onActivate = createAsyncThunk(
  "accountRecovery/onActivate",
  async ({ patientId }) => {
    const url = `/api/dummy/account-recovery/onActivate/${patientId}`;
    return fetch(url, { method: "POST" }).then((res) => res.json());
  }
);

export const onUnlock = createAsyncThunk(
  "accountRecovery/onUnlock",
  async ({ patientId }) => {
    const api = new Api();
    const url = `/ecp/accountRecovery/unlockAccountByAdmin/${patientId}`;

    try {
      const response = await api.getResponse(url, null, "get");

      return {
        success: true,
        response,
      };
    } catch (error) {
      console.error({ error });
      return {
        success: false,
        response: error,
      };
    }
  }
);

export const onSendPasswordReset = createAsyncThunk(
  "accountRecovery/onSendPasswordReset",
  async ({ patientId, patientData, selectedCommunication }) => {
    const api = new Api();
    const url = `/ecp/accountRecovery/sendPasswordResetLinkToUser`;
    const domain = window.location.origin;
    const NEXT_PUBLIC_ONE_TIME_LINK = process.env.NEXT_PUBLIC_ONE_TIME_LINK;
    const postBody = {
      patient: {
        userName: patientData.username,
      },
      preferredComunication: selectedCommunication,
      resetPassword: true,
      link: `${domain}${NEXT_PUBLIC_ONE_TIME_LINK}`,
    };

    try {
      const response = await api.getResponse(url, postBody, "post");

      return {
        success: true,
        response,
      };
    } catch (error) {
      console.error({ error });
      return {
        success: false,
        response: error,
      };
    }
  }
);

export const onShareUsername = createAsyncThunk(
  "accountRecovery/onShareUsername",
  async ({ patientId, selectedCommunication }) => {
    const api = new Api();
    const url = `/ecp/accountRecovery/shareUserNameByAdmin`;
    const postBody = {
      patient: {
        id: patientId,
      },
      preferredComunication: selectedCommunication,
    };

    try {
      const response = await api.getResponse(url, postBody, "post");

      return {
        success: true,
        response,
      };
    } catch (error) {
      console.error({ error });
      return {
        success: false,
        response: error,
      };
    }
  }
);

export const onViewSecurityQuestions = createAsyncThunk(
  "accountRecovery/onViewSecurityQuestions",
  async ({ patientId }) => {
    const api = new Api();
    const url = `/ecp/accountRecovery/viewSecurityQuestions/${patientId}`; // d3724cd1-ebae-4f1a-82a5-544ff33b0313

    try {
      const response = await api.getResponse(url, null, "get", false);

      return {
        success: true,
        response,
      };
    } catch (error) {
      console.error({ error });
      return {
        success: false,
        response: error,
      };
    }
  }
);

const getPatientStatus = (data) => {
  if (data.verified != "active") return "inactive";
  switch (data.status) {
    case "locked":
    case "L":
      return "locked";
    case "unlocked":
    default:
      return "unlocked";
  }
};

const buildAccountData = (data) => {
  return data.map((item) => {
    return {
      ...item,
      name: item.patientName || item.name,
      email: item.emailId,
      phone: item.phoneNumber,
      status: getPatientStatus(item),
    };
  });
};

const INITIAL_STATE = {
  patientList: [],
  securityQuestions: [],
  securityQuestionsRaw: {},
  status: null,
  error: null,
};

export const accountRecoveryStore = createSlice({
  name: "index",
  initialState: INITIAL_STATE,
  reducers: {
    setStatus: (state, { payload }) => {
      state.status = payload;
    },
    setAccountDataById: (state, { payload }) => {
      const updatedState = state.patientList.map((item, idx) => {
        const id = item.id || item.patientId;
        const payloadId = payload.id || payload.patientId;
        if (payloadId === id) {
          item = payload;
        }
        return item;
      });

      state.patientList = buildAccountData(updatedState);
    },
  },
  extraReducers: {
    [fetchPatientAccount.pending]: (state) => {
      state.status = "loading";
    },
    [fetchPatientAccount.fulfilled]: (state, { payload }) => {
      state.patientList = buildAccountData(payload);
      state.status = "success";
    },
    [fetchPatientAccount.rejected]: (state, action) => {
      state.error = action;
      state.status = "failed";
    },
    [onViewSecurityQuestions.fulfilled]: (state, { payload }) => {
      const response = payload.response;
      let mapped = [];
      if (response?.ResponseCode === 3000) {
        const securityQuestions = response.SecurityQuestions[0];
        mapped = Object.keys(securityQuestions).map((keys, item) => {
          return {
            question: keys,
            answer: securityQuestions[keys],
          };
        });

        state.securityQuestionsRaw = response.SecurityQuestions;
      }

      state.securityQuestions = mapped;
      state.status = "success";
    },
    [fetchPatientAccount.rejected]: (state, action) => {
      state.error = action;
      state.securityQuestions = [];
    },
  },
});

// Action creators are generated for each case reducer function
export const { setAccountDataById, setStatus } = accountRecoveryStore.actions;

export default accountRecoveryStore.reducer;
