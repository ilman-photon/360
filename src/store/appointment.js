import { faker } from "@faker-js/faker";
import { createSlice } from "@reduxjs/toolkit";

const DEFAULT_PROVIDER_INFO_DATA = {
  providerId: 0,
  name: null,
  address: null,
  rating: null,
  phoneNumber: null,
  distance: null,
  image: null,
  from: null,
  to: null,
  location: {
    latitude: 0,
    longitude: 0,
  },
};

export const DEFAULT_PATIENT_INFO_DATA = {
  name: null,
  firstName: "",
  lastName: "",
  dob: null,
  phoneNumber: null,

  email: "",
  password: "",
  preferredCommunication: "both",
};

const DEFAULT_APPOINTMENT_INFO_DATA = {
  appointmentType: null,
  date: null,
  insuranceCarrier: null,
};

const DEFAULT_USER_SCHEDULE_APPOINTMENT_DATA = {
  providerInfo: DEFAULT_PROVIDER_INFO_DATA,
  patientInfo: DEFAULT_PATIENT_INFO_DATA,
  appointmentInfo: DEFAULT_APPOINTMENT_INFO_DATA,
};

const DEFAULT_FILTER_DATA = {
  date: new Date(),
  location: "",
  insuranceCarrier: "",
  purposeOfVisit: "",
};

const appointmentSlice = createSlice({
  name: "appointment",
  initialState: {
    appointmentSchedule: DEFAULT_USER_SCHEDULE_APPOINTMENT_DATA,
    filterData: DEFAULT_FILTER_DATA,
    isFilterApplied: false,
    providerListData: [],
  },
  reducers: {
    setAppointmentSchedule: (state, { payload }) => {
      state.appointmentSchedule = payload;
    },
    resetAppointmentSchedule: (state) => {
      state.appointmentSchedule = DEFAULT_USER_SCHEDULE_APPOINTMENT_DATA;
      state.isFilterApplied = false;
      state.providerListData = [];
    },
    editAppointmentScheduleData: (state, { payload }) => {
      state.appointmentSchedule[payload.key] = payload.value;
    },
    setDummyAppointmentSchedule: (state) => {
      state.appointmentSchedule = {
        providerInfo: {
          providerId: 0,
          name: "Dr. Sonha Nguyen",
          address: {
            addressLine1: "673 Herzog Locks", // faker.address.streetAddress(),
            addressLine2: "Suite 300", // faker.address.secondaryAddress(),
            city: "New York", // faker.address.city(),
            state: "NY", // faker.address.stateAbbr(),
            zipcode: "53891", // faker.address.zipCode("#####"),
          },
          rating: 5, //faker.helpers.arrayElement[(1, 2, 3, 4, 5)],
          phoneNumber: "(123) 123-4567", // faker.phone.number("(###) ###-####"),
          distance: "10 mi",
          image: faker.image.imageUrl(275, 173),
          from: new Date(),
          to: new Date(),
          location: {
            latitude: 41.881832, // faker.address.latitude(),
            longitude: -87.623177, // faker.address.longitude(),
          },
        },
        patientInfo: DEFAULT_PATIENT_INFO_DATA,
        appointmentInfo: {
          appointmentType: "Eye Exam",
          date: "2022-08-29T11:12:55.367Z",
          insuranceCarrier: "RSUD",
        },
      };
    },
    setFilterData: (state, { payload }) => {
      state.filterData = payload;
      state.appointmentSchedule.appointmentInfo = {
        ...state.appointmentSchedule.appointmentInfo,
        appointmentType: payload.purposeOfVisit,
        insuranceCarrier: payload.insuranceCarrier,
      };
    },
    resetFilterData: (state) => {
      state.filterData = DEFAULT_FILTER_DATA;
    },
    setIsFilterApplied: (state, { payload }) => {
      state.isFilterApplied = payload;
    },
    setProviderListData: (state, { payload }) => {
      state.providerListData = payload;
    },
  },
});

// Action creators are generated for each case reducer function
export const {
  setAppointmentSchedule,
  resetAppointmentSchedule,
  editAppointmentScheduleData,
  setDummyAppointmentSchedule,
  setFilterData,
  resetFilterData,
  setIsFilterApplied,
  setProviderListData,
} = appointmentSlice.actions;

export default appointmentSlice.reducer;
