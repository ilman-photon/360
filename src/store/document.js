import { createSlice, createAsyncThunk } from "@reduxjs/toolkit";
import { Api } from "../pages/api/api";

export const fetchDocuments = createAsyncThunk(
  "document/fetchDocuments",
  async ({ patientId, category }) => {
    let url;
    let categoryId;
    if (category === "test-lab-result") {
      url = `/ecp/testResult/${patientId}`;
    } else {
      switch (category) {
        case "insurance-documents":
          break;
        case "health-record":
          break;
        case "care-plan-overview":
          categoryId = "Medical-Record";
          break;
        default:
          categoryId = "Intake-Forms";
          break;
      }
      url = `/ecp/patient/getPatientDocumentByCategory/${patientId}/documents?pageSize=10&pageNo=0&sortBy=updated&sortOrder=dsc&search.query=((category=eq=${categoryId}))`;
    }
    const api = new Api();
    return api.getResponse(url, null, "get");
  }
);

const documentSlice = createSlice({
  name: "document",
  initialState: {
    documentList: [],
    status: null,
  },
  extraReducers: {
    [fetchDocuments.pending]: (state) => {
      state.status = "loading";
    },
    [fetchDocuments.fulfilled]: (state, { payload }) => {
      state.documentList = payload.entities;
      state.status = "success";
    },
    [fetchDocuments.rejected]: (state) => {
      state.status = "failed";
    },
  },
});

export default documentSlice.reducer;
