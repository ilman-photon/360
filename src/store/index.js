import { createSlice } from "@reduxjs/toolkit";

const INITIAL_STATE = {
  loading: true,
  counter: 0,
};

export const indexStore = createSlice({
  name: "index",
  initialState: INITIAL_STATE,
  reducers: {
    setLoading: (state, action) => {
      state.loading = action.payload;
    },
    increment: (state, action) => {
      state.counter++;
    },
  },
});

// Action creators are generated for each case reducer function
export const { setLoading, increment } = indexStore.actions;

export default indexStore.reducer;
