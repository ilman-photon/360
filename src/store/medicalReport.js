import { createSlice, createAsyncThunk } from "@reduxjs/toolkit";

export const fetchTestLabResult = createAsyncThunk(
  "medical-report/fetchTestLabResult",
  async () => {
    return fetch("/api/dummy/medical-report/test-lab-result").then((res) =>
      res.json()
    );
  }
);

export const fetchCarePlan = createAsyncThunk(
  "medical-report/fetchCarePlan",
  async () => {
    return fetch("/api/dummy/medical-report/test-lab-result").then((res) =>
      res.json()
    );
  }
);

const medResSlice = createSlice({
  name: "medicalResult",
  initialState: {
    testLabData: [],
    carePlanData: [],
    status: null,
  },
  extraReducers: {
    [fetchTestLabResult.pending]: (state) => {
      state.status = "loading";
    },
    [fetchTestLabResult.fulfilled]: (state, { payload }) => {
      state.testLabData = payload;
      state.status = "success";
    },
    [fetchTestLabResult.rejected]: (state) => {
      state.status = "failed";
    },
    [fetchCarePlan.pending]: (state) => {
      state.status = "loading";
    },
    [fetchCarePlan.fulfilled]: (state, { payload }) => {
      state.carePlanData = payload;
      state.status = "success";
    },
    [fetchCarePlan.rejected]: (state) => {
      state.status = "failed";
    },
  },
});

export default medResSlice.reducer;
