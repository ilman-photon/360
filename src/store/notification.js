import { createSlice, createAsyncThunk } from "@reduxjs/toolkit";

export const fetchNotifications = createAsyncThunk(
  "user/fetchNotifications",
  async () => {
    return fetch("/api/dummy/notification").then((res) => res.json());
  }
);

const INITIAL_STATE = {
  list: [],
  status: "loading",
};

export const notificationStore = createSlice({
  name: "index",
  initialState: INITIAL_STATE,
  reducers: {
    markAllAsRead: (state) => {
      state.list = state.list.map((v) => {
        return {
          ...v,
          isRead: true,
        };
      });
    },
  },
  extraReducers: {
    [fetchNotifications.pending]: (state) => {
      state.status = "loading";
    },
    [fetchNotifications.fulfilled]: (state, { payload }) => {
      state.list = payload;
      state.status = "success";
    },
    [fetchNotifications.rejected]: (state) => {
      state.status = "failed";
    },
  },
});

// Action creators are generated for each case reducer function
export const { markAllAsRead } = notificationStore.actions;

export default notificationStore.reducer;
