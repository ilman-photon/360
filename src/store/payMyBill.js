import { createSlice } from "@reduxjs/toolkit";

const calculateBalanceObject = (summaryData) => {
  return (
    summaryData.totalRetail -
    summaryData.totalInsurance -
    summaryData.totalDiscount +
    summaryData.totalTax -
    summaryData.totalAdjustment -
    summaryData.totalPayment
  );
};

const getObjectData = (item) => ({
  invoiceNumber: item?._invoiceNumber,
  dos: item?.serviceDate,
  creditBalance: calculateBalanceObject(item.summary),
  patientDue: calculateBalanceObject(item.summary),
  provider: `${item?.provider?.firstName} ${item?.provider?.lastName}`,
  invoiceType: item?._invoiceType,
  status: item?._invoiceStatus,
  id: item?._id,
  patient: item?.patient,
});

const getDescription = (item) => {
  let desc = "";
  item.map((itemDesc) => {
    if (item.length > 1) {
      desc = `${desc}, ${itemDesc.category}`;
    } else {
      desc = itemDesc.category;
    }
  });
  return desc;
};

const getSummaryObjectData = (item) => ({
  id: item?._id,
  invoiceNumber: item?._invoiceNumber,
  dos: item?.serviceDate,
  balanceDue: 75,
  totalCharges: 227.5,
  totalAllowed: 239.5,
  insurancePaid: 174.5,
  patientPaid: 302.2,
  description: getDescription(item.lineItems),
  providerName: `${item?.provider?.firstName} ${item?.provider?.lastName}`,
});

const pushNewDataByCondition = (newData, item, isOpen) => {
  if (isOpen) {
    if (
      item?._invoiceStatus == "OPEN" ||
      item?._invoiceStatus == "IN_PROGRESS"
    ) {
      newData.push(getObjectData(item));
    }
  } else if (item?._invoiceStatus == "FINALIZED") {
    newData.push(getObjectData(item));
  }
};

export const mappingListData = (data, isOpen) => {
  let newData = [];
  data?.entities.map((item) => {
    pushNewDataByCondition(newData, item, isOpen);
  });
  return newData;
};

const billSlice = createSlice({
  name: "bill",
  initialState: {
    billingOpenList: [],
    billingHistoryList: [],
    summaryData: {},
    searchDataList: [],
    status: "loading",
  },
  reducers: {
    setStatus: (state, { payload }) => {
      state.status = payload;
    },
    setBillingOpenList: (state, { payload }) => {
      state.billingOpenList = mappingListData(payload.response, payload.isOpen);
    },
    setBillingHistoryList: (state, { payload }) => {
      state.billingHistoryList = mappingListData(
        payload.response,
        payload.isOpen
      );
    },
    setSearchDataList: (state, { payload }) => {
      state.searchDataList = mappingListData(payload.response, payload.isOpen);
    },
    setSummaryData: (state, { payload }) => {
      state.summaryData = getSummaryObjectData(payload.response);
    },
  },
});

// Action creators are generated for each case reducer function
export const {
  setStatus,
  setBillingOpenList,
  setBillingHistoryList,
  setSearchDataList,
  setSummaryData,
} = billSlice.actions;

export default billSlice.reducer;
