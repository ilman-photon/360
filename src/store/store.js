import { configureStore } from "@reduxjs/toolkit";
import index from "../store/index";
import user from "../store/user";

export const store = configureStore({
  reducer: {
    user,
    index,
  },
});

export default store;
