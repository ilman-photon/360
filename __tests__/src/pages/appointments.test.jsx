import Appointments from "../../../src/pages/patient/appointments";
import "@testing-library/jest-dom";
import { act, render, waitFor } from "@testing-library/react";
import App from "next/app";
import axios from "axios";
import MockAdapter from "axios-mock-adapter";
import { Provider } from "react-redux";
import store from "../../../src/store/store";
import Cookies from "universal-cookie";

jest.mock("universal-cookie", () => {
  class MockCookies {
    static result = {};
    get() {
      return MockCookies.result;
    }
    remove() {
      return jest.fn();
    }
  }
  return MockCookies;
});

describe("Render Appointment", () => {
  let container;
  const mock = new MockAdapter(axios);
  const userData = {
    appointmentList: [
      {
        appointmentId: "1",
        providerInfo: {
          providerId: "1",
          name: "Paul Wagner Md",
          position: "Scripps Eyecare",
          address: {
            addressLine1: "51 West 51st Street",
            addressLine2: "Floor 3, Suite 320 Midtown",
            city: "Florida",
            state: "FR",
            zipcode: "54231",
          },
          rating: "5",
          phoneNumber: "8572999989",
          distance: "10 mi",
          image: "/doctor.png",
          from: "2022-07-18",
          to: "2022-07-23",
          location: {
            latitude: 32.751204,
            longitude: -117.1641166,
          },
        },
        patientInfo: {
          name: "Rebecca Chan",
          firstname: "Rebecca",
          lastname: "Chan",
          dob: "12/12/2022",
          phoneNumber: "1234567890",
        },
        appointmentInfo: {
          appointmentType: "Eye Exam",
          date: "Thu, 12 Jan 2023 04:30:00 EST",
          insuranceCarrier: ["ECP Vision", "BlueCare Vision"],
        },
      },
      {
        appointmentId: "1",
        providerInfo: {
          providerId: "1",
          name: "Dr. Sonha Nguyen",
          position: "Scripps Eyecare",
          address: {
            addressLine1: "51 West 51st Street",
            addressLine2: "Floor 3, Suite 320 Midtown",
            city: "Florida",
            state: "FR",
            zipcode: "54231",
          },
          rating: "5",
          phoneNumber: "8572999989",
          distance: "10 mi",
          image: "/doctor.png",
          from: "2022-07-18",
          to: "2022-07-23",
          location: {
            latitude: 32.751204,
            longitude: -117.1641166,
          },
        },
        patientInfo: {
          name: "Rebecca Chan",
          firstname: "Rebecca",
          lastname: "Chan",
          dob: "12/12/2022",
          phoneNumber: "1234567890",
        },
        appointmentInfo: {
          appointmentType: "Eye Exam",
          date: "Thu, 12 Jan 2023 04:30:00 EST",
          insuranceCarrier: ["ECP Vision", "BlueCare Vision"],
        },
      },
    ],
  };
  beforeEach(async () => {
    Cookies.result = { authorized: true };
    mock
      .onGet(
        `${window.location.origin}/api/dummy/appointment/my-appointment/getAllAppointment/98f9404b-6ea8-4732-b14f-9c1a168d8066`
      )
      .reply(200, userData);
    act(() => {
      container = render(
        <Provider store={store}>
          {Appointments.getLayout(<Appointments />)}
        </Provider>
      );
    });
    await waitFor(() => {
      container.getAllByText(userData.appointmentList[0].patientInfo.name);
    });
  });

  afterAll(() => {
    // mock.reset();
  });

  test("is Appointment page render", async () => {
    expect(
      container.getAllByText(userData.appointmentList[0].patientInfo.name)[0]
    ).toBeInTheDocument();
  });
});
