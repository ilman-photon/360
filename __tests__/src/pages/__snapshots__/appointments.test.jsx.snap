// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Appointment is Appointment page render 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="defaultLayout"
      >
        <header
          class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionRelative css-1gg3l1a-MuiPaper-root-MuiAppBar-root"
        >
          <div
            class="MuiContainer-root MuiContainer-maxWidthXl css-19r6kue-MuiContainer-root"
          >
            <div
              class="MuiToolbar-root MuiToolbar-regular css-r6ewbb-MuiToolbar-root"
            >
              <span
                style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
              >
                <span
                  style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
                >
                  <img
                    alt=""
                    aria-hidden="true"
                    src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27124%27%20height=%2736%27/%3e"
                    style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                  />
                </span>
                <img
                  aria-label="Clarkson Eyecare logo"
                  data-nimg="intrinsic"
                  decoding="async"
                  role="img"
                  src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                  style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                  tabindex="0"
                />
                <noscript />
              </span>
            </div>
          </div>
        </header>
        <div
          class="defaultContainer"
        >
          <div
            class="container MuiBox-root css-0"
          >
            <div
              aria-label="Appointments heading"
              class="titleHeadingWrapper"
              style="position: relative;"
              tabindex="0"
            >
              <div
                aria-hidden="true"
                class="title"
              >
                Appointments
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="defaultLayout"
    >
      <header
        class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionRelative css-1gg3l1a-MuiPaper-root-MuiAppBar-root"
      >
        <div
          class="MuiContainer-root MuiContainer-maxWidthXl css-19r6kue-MuiContainer-root"
        >
          <div
            class="MuiToolbar-root MuiToolbar-regular css-r6ewbb-MuiToolbar-root"
          >
            <span
              style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
            >
              <span
                style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
              >
                <img
                  alt=""
                  aria-hidden="true"
                  src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27124%27%20height=%2736%27/%3e"
                  style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                />
              </span>
              <img
                aria-label="Clarkson Eyecare logo"
                data-nimg="intrinsic"
                decoding="async"
                role="img"
                src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                tabindex="0"
              />
              <noscript />
            </span>
          </div>
        </div>
      </header>
      <div
        class="defaultContainer"
      >
        <div
          class="container MuiBox-root css-0"
        >
          <div
            aria-label="Appointments heading"
            class="titleHeadingWrapper"
            style="position: relative;"
            tabindex="0"
          >
            <div
              aria-hidden="true"
              class="title"
            >
              Appointments
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
